crates/miden-objects/src/testing/account_component.rs

```diff
@@ -68,3 +68,112 @@ impl From<AccountMockComponent> for AccountComponent {
             .with_supports_all_types()
     }
 }
+
+// MOCK AUTH COMPONENT
+// ================================================================================================
+
+/// Creates a mock authentication [`AccountComponent`] for testing purposes.
+///
+/// The component defines an `auth__basic` procedure that always increments the nonce by 1.
+pub struct MockAuthComponent {
+    library: Library,
+}
+
+impl MockAuthComponent {
```

Comment:
I wonder if `IncrNonceAuthComponent` or something would be more descriptive, since `Mock` doesn't really say that much. `NoopAuthComponent` for instance, is very descriptive, but I don't have a great alternative naming suggestion for `MockAuthComponent`.